### DISCUSSION TITLE
What should this CAN stack be called?

### DISCUSSION THREAD

**Discussion by ad3154** [Open-Agriculture/AgIsoStack-plus-plus#30] [Q&A] [answered]
I am in the process of developing a "getting started" and tutorial website based on sphinx! My goal is to have something similar to projects like this: https://docs.ros.org/en/rolling/

But, I am getting a bit tired of just calling it "the ISO 11783 CAN stack". Though descriptive, I was wondering if anyone had a good idea for a shorter, catchier name. Branding is kind of important for an open source project.

Suggestions are welcome. Please keep suggestions compliant with the project's [Code of conduct](https://github.com/ad3154/ISO11783-CAN-Stack/blob/3c5662340df70dd472a96557926af6ea0109465c/CODE_OF_CONDUCT.md)

**Comment by Thom-de-Jong** [comment]
I think "OpenIsobus" would be catchy.

**Comment by ad3154** [comment, marked as answer]
It seems like OpenIsobus might be somewhat taken already by this company https://anedo.eu/openisobus/

I am somewhat partial to ISOBUS++ myself

**Reply by GwnDaan** [reply]
ISOBUS++ fits particularly well with C++ so I like it. I had ISOBUSLib in mind, but I don't like the way it's similar to IsoAgLib. So I prefer ISOBUS++ instead

**Reply by ad3154** [reply]
![image](https://user-images.githubusercontent.com/10929341/208755198-36775044-70b2-4dad-bd2b-119e33060afe.png)
Tentative branding 😯

**Comment by dlips** [comment]
Hello everyone,
I know this is an older topic and you already renamed the project, but I want to raise my concerns with both proposals. Isobus is a registered trademark in the EU by the AEF (probably also in the US), which makes projects with the suggested names vulnerable to any legal action by the trademark holders as they are to similiar and in the same field. This gives me stomach pain, while just started preparing a contribution for the project, I think about deleting my forked repo because of that. In principle everything can go well, but please keep in mind that there are also software companies in the AEF that provide ISO 11783 stacks as a product, who could see this open source project as a competion at some point.
As I like the current name really much, I think it is also kind of dangerous.

**Reply by ad3154** [reply]
I think that is a fair concern.
I was hoping it would not be an issue for a number of reasons:

- This project is not commercial
- We do not use any AEF branding, logos, the actual ISOBUS icon, etc (though maybe we can be more explicit about being unaffiliated)
- Our logo always contains the tag line "The Free ISO11783 CAN Stack" to distinguish itself and is stylized in lowercase instead of uppercase without any AEF brand colours
- It's basically just descriptive of what it implements, like how this: https://github.com/ttlappalainen/NMEA2000 is just an implementation of portions of NMEA2k

But, I do agree it could be an issue. I will attempt to reach out to the AEF to seek their opinion on if our name is acceptably distant from their trademark or if we might need to change it.

**Reply by dlips** [reply]
I am not so familiar with the legal details of trademark, but I think it is a good idea to be pro-active about this topic. I know there is also the "Competance Center ISOBUS" which uses ISOBUS in its name but is in principle distinct from the AEF. However, they are themselves member of the AEF.

**Reply by ad3154** [reply]
Yeah, agreed. I have sent them an email and will let you know when I hear something back.

In my perfect world, our project would be part of the AEF, haha, but I do not know how realistic that is.

**Reply by ad3154** [reply]
I did get a response from the general manager of AEF asking us to change our name, so as unfortunate as that is, we will have to choose another name.

**Reply by ad3154** [reply]
I'll pitch some more names, but I am not very creative with names... feel free to add suggestions.

- IsoStack++ (I like keeping Iso in there)
- AgroCAN++
- AgriCAN++
- AgIsoEngine
- ??

We could distinguish variants by language and use, like in Rust we could be nameRS instead of name++ or something, haha.

**Reply by GwnDaan** [reply]
Hmm, that's really unfortunate... Let's all think of another name, that is informative and still catchy.

I'm thinking of the following keywords that match this library/stack well:
- Agriculture / Ag
- CAN
- ISO
- Open source
- C++

Possible names I can think of:
- AgIsoCAN ->  (AgIsoCAN++, AgIsoCAN-rs)
- AgOpenCAN -> (AgOpenCAN++, AgOpenCAN-rs)

In the end a lot of variantions are possible with these words. And after a few minutes of pronouncing combinations out loud, they all started to sound the same to me😂

While we are at it, I have also a few suggestions for a possible github orginazation:
- OpenAgriculture / OpenAg

**Reply by Thom-de-Jong** [reply]
Really unfortunate as the word best describing the project simply is: Isobus...

I'm not a fan of calling it any combination of the words Open / CAN as it might confuse to much with the CANOpen protocol or even just the CAN bus in general. (This project is more specific than just CAN)

I don't know of any new names on the top of my head but i like IsoStack++, IsoStack-rs, etc. :stuck_out_tongue:

@GwnDaan, I like the organisation name OpenAgriculture, makes growth into other things than just Isobus possible while perfectly describing our combined goal.

**Reply by dlips** [reply]
+1 for IsoStack. 

Other ideas: AgNet, AgriNet, CANnector, CANagTor, IsoCANCore, AgroCore, AgriLink, AgroWire, BarnBytes, farmline, farmwire, transfarm, isolink (sorry, some are silly, just brain storming)

**Reply by GwnDaan** [reply]
Great stuff! I'd like to add:
- AgIsoStack -> (AgIsoStack++, AgIsoStack-rs, etc.) 

I agree with the arguments @Thom-de-Jong made regarding CAN. But I'd like to add that ISO is also much broader than only ISO11783/isobus. Hence calling it solely "IsoStack" might not really be that descriptive of what this project really is. So I feel like adding agriculture / tractor / ag-machinery or something is a must for that reason. Since there are plans of having (parts of) isobus **not** use CAN communication anymore in the future, I think we also should maybe not include CAN in the name. 

This is the abstraction of ISO11783 from [iso.org](https://www.iso.org/standard/57556.html), maybe it might help:
>ISO 11783 as a whole specifies a serial data network for control and communications on forestry or agricultural tractors and mounted, semi-mounted, towed or self-propelled implements.

**Reply by ad3154** [reply]
I also think tying it back to agriculture is probably a good idea, as is avoiding confusion with CANopen. Technically the target audience is broader with forestry and construction being other industry groups that use the standard, but Ag is of course the largest and most invested industry. Plus, most of the attention this project seems to get comes from the Ag industry. So I think I am liking "AgIsoStack" the best so far... and yeah I also agree "OpenAgriculture" is great as an Org name (we have been searching for a good Org name for a while, haha)

I think that no matter what we go with we'll want to keep mentioning the ISOBUS standard, as well as J1939 in our description and README as well, which will help people find us using searches both on and off GitHub. Though it may be worth adding some language to the README that we are unaffiliated with AEF and SAE.

**Reply by ad3154** [reply]
I have created an Org here https://github.com/Open-Agriculture and will move this repository into it after making an announcement here in the discussions pages
